src/ancillary.rs

```diff
@@ -304,22 +303,23 @@ pub enum AncillaryItem<'a> {
 }
 
 /// An iterator over ancillary messages received with `recv_ancillary()`.
-pub struct Ancillary<'a> {
+pub struct Ancillary {
     // addr and bytes are not used here:
     // * addr is usually placed on the stack by the calling wrapper method,
     //   which means that its lifetime ends when this struct is returned.
     // * the iovec is incremented by Linux, but possibly not others.
     msg: msghdr,
 
-    _ancillary_buf: PhantomData<&'a[u8]>,
```

Comment:
The `PhantomData` is there to prevent this struct from outliving the buffer that `next_message` points into.

Reply:
Wasn't needed anymore since all data was copied, but now that we don't allocate a `Vec` I brought it back

Reply:
`AncillaryItem` is just the returned items.
The items not yet returned by the iterator are still stored in a buffer which `msg` and `next_message` points into.
Therefore `Ancillary` would still need to have a lifetime to that buffer, even if the returned items don't point to it.